tools/api-key-rotation/api_key_rotation_checker/main.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python3
+
+#
+# Copyright 2021 Google LLC.
+# This software is provided as is, without warranty or representation for any use or purpose. # pylint: disable = line-too-long
+# Your use of it is subject to your agreement with Google.
+#
+
+import subprocess
+import requests # pylint: disable=import-error
+import googleapiclient.discovery # pylint: disable=import-error
+from datetime import datetime, timedelta
+from dateutil import parser as date_parser # pylint: disable=import-error
+from google.api_core import exceptions # pylint: disable=import-error
+from googleapiclient.errors import HttpError # pylint: disable=import-error
+from googleapiclient.discovery_cache.base import Cache # pylint: disable=import-error
+import logging
+import sys
+import argparse
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json # pylint: disable=import-error
+
+def x_days_ago(rotation_period):
+    """
+    Get the date x days ago to use for our comparison.
+
+    Args:
+
+    rotation_period - The rotation period in days (default is 90)
+    """
+
+    rotation_days = rotation_period
+    today = datetime.now()
+
+    # Find the date x days ago
+    rotation_date = today - timedelta(days=rotation_days)
+    logging.info("%s days ago was %s \n", rotation_days, rotation_date)
+
+    return rotation_date
+
+def create_project_list(service):
+    """
+    Get our project list from GCP.
+
+    Args:
+
+    service - A resource manager service created via create_service() function
+    """
+
+    logging.info(
+        "Getting GCP project list. This may take a minute.. \n")
+
+    # Collect all the projects
+    projects = []
+
+    # Paginate through the list of all available projects
+    request = service.projects().list()
+    while request is not None:
+        try:
+            response = request.execute()
+            projects.extend(response.get("projects", []))
+            request = service.projects().list_next(request, response)
+        except HttpError as err:
+            logging.error(err)
```

Comment:
This while loop can become an infinite loop. What happens if the request fails consistently?

Reply:
That's true and it's not really necessary to have the while loop. I removed it for now.